commit b99c72e4885397150a463b86150438e2843e7f72 (refs/changes/02/3902/5)
Author: Kody A Kantor <kody@kkantor.com>
Date:   2018-05-10T21:11:38+00:00 (1 year, 5 months ago)
    
    MORAY-462 artedi metrics should report shard name
    Reviewed by: Cody Peter Mello <melloc@writev.io>
    Approved by: Cody Peter Mello <melloc@writev.io>

diff --git a/etc/config.coal.manta.json b/etc/config.coal.manta.json
index 5790d98..c5e22ba 100644
--- a/etc/config.coal.manta.json
+++ b/etc/config.coal.manta.json
@@ -33,5 +33,6 @@
     "expiry": 60
   },
   "server_uuid": "fake-manta-coal-server-uuid",
-  "datacenter": "coal"
+  "datacenter": "coal",
+  "service_name": "1.moray.coal.joyent.us"
 }
diff --git a/lib/server.js b/lib/server.js
index b7e29a7..6ea20c7 100644
--- a/lib/server.js
+++ b/lib/server.js
@@ -68,6 +68,11 @@ function MorayServer(options) {
         pid: process.pid
     };
 
+    /* service_name includes the shard name in Manta deployments. */
+    if (options.service_name) {
+        labels.service = options.service_name;
+    }
+
     var collector = mod_artedi.createCollector({
         labels: labels
     });
diff --git a/package.json b/package.json
index a278751..6dc39a9 100644
--- a/package.json
+++ b/package.json
@@ -7,7 +7,7 @@
     "main": "lib/index.js",
     "dependencies": {
         "ajv": "4.11.4",
-        "artedi": "1.1.1",
+        "artedi": "1.3.0",
         "assert-plus": "1.0.0",
         "bunyan": "1.8.10",
         "bunyan-syslog": "0.2.2",
diff --git a/sapi_manifests/moray/template b/sapi_manifests/moray/template
index 00b9609..ea0fc5e 100644
--- a/sapi_manifests/moray/template
+++ b/sapi_manifests/moray/template
@@ -34,5 +34,6 @@
     "expiry": 60
   },
   "server_uuid": "{{auto.SERVER_UUID}}",
-  "datacenter": "{{DATACENTER}}"
+  "datacenter": "{{DATACENTER}}",
+  "service_name": "{{SERVICE_NAME}}"
 }
